### DISCUSSION TITLE
domaincheck won't start

### DISCUSSION THREAD

**Discussion by orfins** [nextcloud/all-in-one#3493] [Questions] [answered]
I'm using the `compose.yaml` from the root directory.
After the initial run `sudo docker compose up`, I can only see `nextcloud-aio-mastercontainer` in `sudo docker container ls -a`.
After I log in to the web interface, I'm redirected to `https://localhost:8080/containers`, and presented with the error `Domaincheck container is not running`.

Checking `sudo docker container ls -a` again, I can now see that `nextcloud-aio-domaincheck` exited after just 1 second:
```
[user@nextcloud]$ sudo docker container ls -a
CONTAINER ID   IMAGE                              COMMAND       CREATED          STATUS                      PORTS                                                                          NAMES
<id>   nextcloud/aio-domaincheck:latest   "/start.sh"   32 seconds ago   Exited (0) 31 seconds ago                                                                                  nextcloud-aio-domaincheck
<id>   nextcloud/all-in-one:latest        "/start.sh"   2 minutes ago    Up 2 minutes (healthy)      0.0.0.0:80->80/tcp, 0.0.0.0:8080->8080/tcp, 0.0.0.0:8443->8443/tcp, 9000/tcp   nextcloud-aio-mastercontainer
```
Running `sudo docker logs nextcloud-aio-domaincheck` shows the error:
`2023-10-06 17:50:32: (../src/network.c.543) socket(): Address family not supported by protocol`

Just a few minutes ago, for some reason, the error was actually:
`permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get ...: dial unix /var/run/docker.sock: connect: permission denied`

**Comment by Zoey2936** [comment]
do you run SE linux?

**Reply by orfins** [reply]
I'm running Fedora which has it on by default IIRC. Is that a problem or is there a way around it?

**Reply by Zoey2936** [reply, marked as answer]
https://github.com/nextcloud/all-in-one#are-there-known-problems-when-selinux-is-enabled just do this

**Reply by orfins** [reply]
@Zoey2936 I added the following to the `.yaml` file:
```
services:
  nextcloud-aio-mastercontainer:
    security_opt:
      - label:disable
    ...
```
The opt is applied to the master container but not to `domaincheck`, and the result is the same:
```
[user@nextcloud]$ sudo docker inspect nextcloud-aio-mastercontainer --format '{{ .Id }}: SecurityOpt={{ .HostConfig.SecurityOpt }}'
<container id>: SecurityOpt=[label:disable]
[user@nextcloud]$ sudo docker inspect nextcloud-aio-domaincheck --format '{{ .Id }}: SecurityOpt={{ .HostConfig.SecurityOpt }}'
<container id>: SecurityOpt=<no value>
[user@nextcloud]$ sudo docker logs nextcloud-aio-domaincheck
2023-10-07 09:18:29: (../src/network.c.543) socket(): Address family not supported by protocol
```

**Reply by Zoey2936** [reply]
Does your server support ipv6?

**Reply by orfins** [reply]
No, it's disabled using `GRUB_CMDLINE_LINUX="ipv6.disable=1 ..."`

**Reply by Zoey2936** [reply]
Then you have found your (seconnd) problem

**Reply by orfins** [reply]
Is it not possible to run `nextcloud` without IPv6?

**Reply by Zoey2936** [reply]
as you see, no

**Reply by orfins** [reply]
I'm not aware of all possible configurations and flags so I couldn't confidently conclude that. Instead, is there a way (and is it even recommended) to disable containers and functionalities I'm not interested in?

**Reply by Zoey2936** [reply]
you could try to skip the domaincheck see the readme
